axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/PurchaseOrderInvoiceServiceImpl.java

```diff
@@ -425,4 +439,192 @@ public PurchaseOrder fillPurchaseOrder(PurchaseOrder purchaseOrder, Invoice invo
+                purchaseOrderLine ->
+                    purchaseOrderLine.getQty().multiply(purchaseOrderLine.getPriceDiscounted()))
+            .reduce(BigDecimal.ZERO, (bd1, bd2) -> bd1.add(bd2));
+
+    BigDecimal percentToInvoice =
+        commonInvoiceService.computeAmountToInvoicePercent(
+            purchaseOrder, amountToInvoice, isPercent, total);
+    AccountConfig accountConfig = accountConfigService.getAccountConfig(purchaseOrder.getCompany());
+    Account advancePaymentAccount = accountConfig.getSupplierAdvancePaymentAccount();
+    Product advancePaymentProduct = accountConfig.getAdvancePaymentProduct();
+
+    if (advancePaymentProduct == null) {
+      throw new AxelorException(
+          purchaseOrder,
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.SO_INVOICE_MISSING_ADVANCE_PAYMENT_PRODUCT));
+    }
+    if (advancePaymentAccount == null) {
+      throw new AxelorException(
+          purchaseOrder,
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(IExceptionMessage.PO_INVOICE_MISSING_SUPPLIER_ADVANCE_PAYMENT_ACCOUNT),
+          purchaseOrder.getCompany().getName());
+    }
+
+    Invoice invoice =
+        createInvoiceAndLines(
+            purchaseOrder,
+            purchaseOrder.getPurchaseOrderLineTaxList(),
+            advancePaymentProduct,
+            percentToInvoice,
+            InvoiceRepository.OPERATION_SUB_TYPE_ADVANCE,
+            advancePaymentAccount);
+
+    if (invoice.getInvoiceLineList() != null) {
+      invoice
+          .getInvoiceLineList()
+          .forEach(
+              invoiceLine -> {
+                invoiceLine.setPurchaseOrderLine(null);
+              });
+    }
+
+    invoice.setPurchaseOrder(purchaseOrder);
+    invoice.setAddressStr(addressService.computeAddressStr(invoice.getAddress()));
+    invoiceService.setDraftSequence(invoice);
+    invoice.setPartnerTaxNbr(purchaseOrder.getSupplierPartner().getTaxNbr());
+
+    return invoiceRepo.save(invoice);
+  }
+
+  protected Invoice createInvoiceAndLines(
+      PurchaseOrder purchaseOrder,
+      List<PurchaseOrderLineTax> taxLineList,
+      Product invoicingProduct,
+      BigDecimal percentToInvoice,
+      int operationSubTypeSelect,
+      Account partnerAccount)
+      throws AxelorException {
+    InvoiceGenerator invoiceGenerator = this.createInvoiceGenerator(purchaseOrder);
+
+    Invoice invoice = invoiceGenerator.generate();
+
+    List<InvoiceLine> invoiceLinesList =
+        (taxLineList != null && !taxLineList.isEmpty())
+            ? this.createInvoiceLinesFromTax(
+                invoice, taxLineList, invoicingProduct, percentToInvoice)
+            : commonInvoiceService.createInvoiceLinesFromOrder(
+                invoice, purchaseOrder.getInTaxTotal(), invoicingProduct, percentToInvoice);
+
+    invoiceGenerator.populate(invoice, invoiceLinesList);
+    // invoice.setAddressStr(purchaseOrder.getMainInvoicingAddressStr());
```

Comment:
Please check why this is commented

Reply:
Removed the line, since there is no main invoicing address in PurchaseOrder